docs/_clef/Rules.md

```diff
@@ -1,26 +1,64 @@
-sort_key: B
+sort_key: C
 ---
 
-The `signer` binary contains a ruleset engine, implemented with [OttoVM](https://github.com/robertkrimen/otto)
+This page provides a fairly low-level explanation for how rules are implemented in 
+Clef. It is a good idea to read the [Introduction to Clef](/docs/clef/introduction)
+and the [Clef tutorial](/docs/clef/tutorial) before diving in to this page.
 
-It enables usecases like the following:
+{:toc}
+-   this will be removed by the toc
+  
+## Introduction
 
-* I want to auto-approve transactions with contract `CasinoDapp`, with up to `0.05 ether` in value to maximum `1 ether` per 24h period
-* I want to auto-approve transaction to contract `EthAlarmClock` with `data`=`0xdeadbeef`, if `value=0`, `gas < 44k` and `gasPrice < 40Gwei`
+Rules in Clef are sets of conditions that determine whether a given action can be
+signed automatically without requiring manual approval from the user. This can be
+useful for automatically approving transactions between a user's own accounts, or
+to approve patterns that are commonly used by applications. This is a UI improvement
+for users making lots of transactions and also a security improvement because transactions
+that do not fit the pattern defined by a ruleset will not be signed unless they are 
+reviewed and manually approved by the end user.
 
-The two main features that are required for this to work well are;
+Rules can define arbitrary conditions such as:
 
-1. Rule Implementation: how to create, manage and interpret rules in a flexible but secure manner
-2. Credential managements and credentials; how to provide auto-unlock without exposing keys unnecessarily.
+* Auto-approve 10 transactions with contract `CasinoDapp`, with value between `0.05 ether` and 
+  `1 ether` per 24h period.
 
-The section below deals with both of them
+* Auto-approve transactions to contract `Uniswapv2` with `value` up to 1 ether, if 
+  `gas < 44k` and `gasPrice < 40Gwei`.
+
+* Auto-approve signing if the data to be signed contains the string `"approve_me"`.
+
+* Auto-approve any requests to list accounts in keystore if the request arrives over IPC
+
+Because the rules are Javascript files they can be customized to implement any arbitrary logic on
+the available request data.
+
+This page will explain how rules are implemented in Clef and how best to manage credentials
+when automatic rulesets are enabled.
 
 ## Rule Implementation
 
-A ruleset file is implemented as a `js` file. Under the hood, the ruleset-engine is a `SignerUI`, implementing the same methods as the `json-rpc` methods
-defined in the UI protocol. Example:
+Although rules in Clef are written in Javascript, Clef itself is written in Go. To
+implement the Javascript rules in Clef, a virtual machine 
+([OttoVM](https://github.com/robertkrimen/otto)) is embedded into the `signer` binary
+that parses and interprets the Javascript code and imposes their logic in Clef. This 
+virtual machine is called the 'ruleset engine'.
```

Comment:
It's not actualy OttoVM any longer. On the whole, I'm sceptical as to whether this paragraph is warranted. Too low level IMO

Reply:
```suggestion
```